js/util/evented.js

```diff
@@ -95,6 +96,18 @@ var Evented = {
     },
 
     /**
+     * Asyncronously fires an event of the specified type. The event will be
+     * fired within a "setTimeout(..., 0)" callback.
+     *
+     * @param {string} type The type of event to fire.
+     * @param {Object} [data] Data to be passed to any listeners.
+     * @returns {Object} `this`
+     */
+    asyncFire: function(type, data) {
+        window.setTimeout(this.fire.bind(this, type, data), 0);
```

Comment:
Well, it's probably technically a race condition -- I don't think the DOM specs guarantee that the `setTimeout(0)` will trigger before the AJAX request onload (which is plausible if, say, the request is immediately satisfied by a cache).